src/api/auth/AuthService.ts

```diff
@@ -0,0 +1,40 @@
+import type { IApiClient } from '../ApiClient';
+
+export interface IAuthApiClient {
+  registerUser(username: string, password: string): Promise<any>;
+}
+
+export class AuthApiClient implements IAuthApiClient {
+  baseUrl: string;
+  authApiClient: IApiClient;
+
+  constructor(authApiClient: IApiClient) {
+    this.baseUrl = 'http://localhost:3000';
```

Comment:
Seems like your API client has already been configured with a base URL. Do you need this one?